Utils/auto_update_docker/extract_affected_content_items.py

```diff
@@ -0,0 +1,447 @@
+                            "pack_path": pack_path,
+                            "docker_image_tag": docker_image_tag,
+                        }
+                    )
+                else:
+                    logging.warning(f"{content_item_py} was returned from the graph, but not found in repo")
+            else:
+                logging.warning(f"{auto_update_docker_image=} configured for {content_item}, skipping")
+    return content_images
+
+
+def docker_tags_parser(benchmark_docker_tags: str) -> dict[str, str]:
+    if not benchmark_docker_tags:
+        return {}
+
+    # Remove spaces, if input has spaces between comma
+    # docker_tags_no_spaces = benchmark_docker_tags.replace(" ", "")
+
+    # Split the input string into key-value pairs
+    pairs = benchmark_docker_tags.split(",")
+
+    # Initialize an empty dictionary to store the key-value pairs
+    result_dict = {}
+
+    # Iterate over each key-value pair
+    for pair in pairs:
+        # Split the pair into key and value
+        key, value = pair.split(":")
+        # strip() to remove spaces
+        result_dict[key.strip()] = value.strip()
+
+    return result_dict
+
+
+@app.command()
+def get_affected_content_items(
+    config_path: str = typer.Option(
+        help="The config file that holds all the configuration of the batches and docker images",
+    ),
+    coverage_report: str = typer.Option(
+        help="The coverage report from last nightly",
+    ),
+    batch_index: int = typer.Option(
+        help="The batch index",
+    ),
+    flow_index: int = typer.Option(
+        help="The flow index",
+    ),
+    benchmark_docker_tags: dict[str, str] = typer.Option(
+        default="",
+        help="Docker images and their benchmark tags, where each docker image will hold the tag from which every content items'"
+        " ",
+        parser=docker_tags_parser,
+    ),
+    docker_images_arg: str = typer.Option(
+        default="ALL",
+        help=(
+            "The docker images that should be affected by the auto update, either a comma"
+            " separated list, the string 'ALL', or 'ALL@docker1,docker2',"
+            " where the last option will exclude the stated docker images"
+        ),
+    ),
+    auto_update_dir: str = typer.Option(
+        default="",
+        help=("The directory that will hold the output files. The default will be the current working directory"),
+    ),
+):
+    # IMPORTANT - "demisto-sdk graph create" must be ran before
+    # Entry point of code
+    path_dir = Path(auto_update_dir) if dir else Path(CWD)
+    if not path_dir.exists():
+        raise Exception(f"{path_dir = } was not found, aborting")
+
+    tests_conf = load_json("Tests/conf.json")
+    # Get nightly packs from tests conf
+    nightly_packs: list[str] = tests_conf.get("nightly_packs", [])
+
+    # Creates flow directory if it does not exist, else it does nothing
+    flow_dir = path_dir / Path(f"flow_{flow_index}")
```

Comment:
whats flow dir and flow index?

Reply:
I explained these in the confluence page